src/Components/Web/src/ElementReferenceExtensions.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components
+{
+    public static class ElementReferenceExtensions
+    {
+        /// <summary>
+        /// Gives focus to an element given its <see cref="ElementReference"/>.
+        /// </summary>
+        /// <param name="elementReference">A reference to the element to focus.</param>
+        /// <returns>The <see cref="ValueTask"/> representing the asynchronous focus operation.</returns>
+        public static ValueTask FocusAsync(this ElementReference elementReference)
+            => elementReference.JSRuntime.InvokeVoidAsync(DomWrapperInterop.Focus, elementReference);
```

Comment:
This is being extra cautious, but could you add a null check for `ElementReference.JSRuntime`? Just to guard against

```
await default(ElementReference).FocusAsync()
```

Reply:
Do you think it would be good to throw an exception if the null check fails, instead of silently not focusing? Or maybe log something?

Reply:
Throwing an exception sounds great.